Miredot documentation fails to build

As already noted in #138: the upgrade to Java 11 required an upgrade of the Miredot plugin to `2.4.1-Java11`. But with that version, the documentation build fails with the following error message:
```
[INFO] --- miredot-plugin:2.4.1-Java11:restdoc (miredot) @ ids.server ---
[WARNING] Rest Framework not explicitly set, using JAX-RS by default.
[INFO] Building reflective model of API and reading annotations...
[ERROR] Your project doesn't contain any REST interfaces, so no output will be generated.
Maybe it is incorrectly configured.
```

This seem to be the same issue as icatproject/icat.server#310.

--- a/pom.xml
+++ b/pom.xml
@@ -159,7 +159,7 @@
             <plugin>
                 <groupId>com.qmino</groupId>
                 <artifactId>miredot-plugin</artifactId>
-                <version>2.4.1-Java11</version>
+                <version>2.4.4-Java11</version>
                 <executions>
                     <execution>
                         <id>miredot</id>
@@ -194,6 +194,9 @@
                         </checks>
                     </analysis>
                     <restModel>
+                        <restFramework>
+                            <name>jakarta</name>
+                        </restFramework>
                         <httpStatusCodes>
                             <httpStatusCode>
                                 <httpCode>400</httpCode>

--- a/src/main/java/org/icatproject/ids/IdsService.java
+++ b/src/main/java/org/icatproject/ids/IdsService.java
@@ -56,6 +56,7 @@ public class IdsService {
      * datafileIds specified along with a sessionId. If two level storage is not
      * in use this has no effect.
      *
+     * @title archive
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds If present, a comma separated list of investigation id values
      * @param datasetIds       If present, a comma separated list of data set id values or
@@ -66,7 +67,6 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws InternalException
      * @throws NotFoundException
-     * @title archive
      * @statuscode 200 To indicate success
      */
     @POST
@@ -84,6 +84,7 @@ public class IdsService {
      * Delete data specified by the investigationIds, datasetIds and datafileIds
      * specified along with a sessionId.
      *
+     * @title delete
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds If present, a comma separated list of investigation id values
      * @param datasetIds       If present, a comma separated list of data set id values or
@@ -95,7 +96,6 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InternalException
      * @throws DataNotOnlineException
-     * @title delete
      * @statuscode 200 To indicate success
      */
     @DELETE
@@ -115,8 +115,8 @@ public class IdsService {
     /**
      * Return the version of the server
      *
-     * @return json string of the form: <samp>{"version":"4.4.0"}</samp>
      * @title Version
+     * @return json string of the form: <samp>{"version":"4.4.0"}</samp>
      */
     @GET
     @Path("version")
@@ -135,6 +135,7 @@ public class IdsService {
      * of which may be omitted, along with a sessionId if the preparedId is not
      * set. If preparedId is set the compress and zip arguments are not used.
      *
+     * @title getData
      * @param preparedId       A valid preparedId returned by a call to prepareData
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds A comma separated list of investigation id values.
@@ -158,7 +159,6 @@ public class IdsService {
      * @throws InternalException
      * @throws InsufficientPrivilegesException
      * @throws DataNotOnlineException
-     * @title getData
      * @statuscode 200 To indicate success
      */
     @GET
@@ -198,6 +198,7 @@ public class IdsService {
      * and datafileIds specified along with a sessionId if the preparedId is not
      * set.
      *
+     * @title getDatafileIds
      * @param preparedId       A valid preparedId returned by a call to prepareData
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds A comma separated list of investigation id values.
@@ -208,7 +209,6 @@ public class IdsService {
      * @throws InternalException
      * @throws NotFoundException
      * @throws InsufficientPrivilegesException
-     * @title getDatafileIds
      * @statuscode 200 To indicate success
      */
     @GET
@@ -248,6 +248,7 @@ public class IdsService {
      * the file system where the IDS is storing data. Only read access to the
      * file is granted.
      *
+     * @title getLink
      * @param sessionId  A valid ICAT session ID
      * @param datafileId the id of a data file
      * @param username   the name of the user who will will be granted access to the
@@ -259,7 +260,6 @@ public class IdsService {
      * @throws InternalException
      * @throws NotFoundException
      * @throws DataNotOnlineException
-     * @title getLink
      * @statuscode 200 To indicate success
      */
     @POST
@@ -278,12 +278,12 @@ public class IdsService {
      * Obtain detailed information about what the ids is doing. You need to be
      * privileged to use this call.
      *
+     * @title getServiceStatus
      * @param sessionId A valid ICAT session ID of a user in the IDS rootUserNames
      *                  set.
      * @return a json string.
      * @throws InternalException
      * @throws InsufficientPrivilegesException
-     * @title getServiceStatus
      * @statuscode 200 To indicate success
      */
     @GET
@@ -298,6 +298,7 @@ public class IdsService {
      * Return the total size of all the data files specified by the
      * investigationIds, datasetIds and datafileIds along with a sessionId.
      *
+     * @title getSize
      * @param preparedId       A valid preparedId returned by a call to prepareData
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds If present, a comma separated list of investigation id values
@@ -309,7 +310,6 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InsufficientPrivilegesException
      * @throws InternalException
-     * @title getSize
      * @statuscode 200 To indicate success
      */
     @GET
@@ -330,6 +330,7 @@ public class IdsService {
      * Return the archive status of the data files specified by the
      * investigationIds, datasetIds and datafileIds along with a sessionId.
      *
+     * @title getStatus
      * @param preparedId       A valid preparedId returned by a call to prepareData
      * @param sessionId        A sessionId returned by a call to the icat server. If the
      *                         sessionId is omitted or null the ids reader account will be
@@ -347,7 +348,6 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InsufficientPrivilegesException
      * @throws InternalException
-     * @title getStatus
      * @statuscode 200 To indicate success
      */
     @GET
@@ -382,12 +382,12 @@ public class IdsService {
      * the end successfully will then go through all those it did not look at
      * because it did not start at the beginning.
      *
+     * @title isPrepared
      * @param preparedId A valid preparedId returned by a call to prepareData
      * @return true if all the data files are ready to be downloaded else false.
      * @throws BadRequestException
      * @throws NotFoundException
      * @throws InternalException
-     * @title isPrepared
      * @statuscode 200 To indicate success
      */
     @GET
@@ -402,8 +402,8 @@ public class IdsService {
      * An ids server can be configured to be read only. This returns the
      * readOnly status of the server.
      *
-     * @return true if readonly, else false
      * @title isReadOnly
+     * @return true if readonly, else false
      * @statuscode 200 To indicate success
      */
     @GET
@@ -417,8 +417,8 @@ public class IdsService {
      * An ids server can be configured to support one or two levels of data
      * storage. This returns the twoLevel status of the server.
      *
-     * @return true if twoLevel, else false
      * @title isTwoLevel
+     * @return true if twoLevel, else false
      * @statuscode 200 To indicate success
      */
     @GET
@@ -431,8 +431,8 @@ public class IdsService {
     /**
      * Should return "IdsOK"
      *
-     * @return "IdsOK"
      * @title ping
+     * @return "IdsOK"
      * @statuscode 200 To indicate success
      */
     @GET
@@ -450,6 +450,7 @@ public class IdsService {
      * by the investigationIds, datasetIds and datafileIds, any of which may be
      * omitted, along with a sessionId.
      *
+     * @title prepareData
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds A comma separated list of investigation id values.
      * @param datasetIds       A comma separated list of data set id values.
@@ -466,7 +467,6 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws NotFoundException
      * @throws InternalException
-     * @title prepareData
      * @statuscode 200 To indicate success
      */
     @POST
@@ -485,6 +485,7 @@ public class IdsService {
     /**
      * Stores a data file
      *
+     * @title put
      * @param body               The contents of the file to be stored
      * @param sessionId          A sessionId returned by a call to the icat server.
      * @param name               A name to assign to the data file
@@ -503,7 +504,6 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws NotImplementedException
      * @throws DataNotOnlineException
-     * @title put
      * @statuscode 201 When object successfully created
      */
     @PUT
@@ -528,6 +528,7 @@ public class IdsService {
      * two fields will be removed shortly as they are only required by the old
      * (GWT based) topcat.
      *
+     * @title putAsPost
      * @param request
      * @return a json object with attributes of "id", "checksum", "location" and
      * "size";
@@ -537,7 +538,6 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws NotImplementedException
      * @throws DataNotOnlineException
-     * @title putAsPost
      * @statuscode 201 When object successfully created
      */
     @POST
@@ -615,6 +615,7 @@ public class IdsService {
      * You must either specify a preparedId or a sessionId. If preparedId is
      * specified then investigationIds, datasetIds and datafileIds are not used.
      *
+     * @title reset
      * @param preparedId       A valid preparedId returned by a call to prepareData
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds A comma separated list of investigation id values.
@@ -625,7 +626,6 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InternalException
      * @throws InsufficientPrivilegesException
-     * @title reset
      * @statuscode 200 To indicate success
      */
     @POST
@@ -647,6 +647,7 @@ public class IdsService {
      * datafileIds specified along with a sessionId. If two level storage is not
      * in use this has no effect.
      *
+     * @title restore
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds If present, a comma separated list of investigation id values
      * @param datasetIds       If present, a comma separated list of data set id values or
@@ -657,7 +658,6 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws InternalException
      * @throws NotFoundException
-     * @title restore
      * @statuscode 200 To indicate success
      */
     @POST
@@ -678,6 +678,7 @@ public class IdsService {
      * storage. If two level storage is not in use this has no
      * effect.
      *
+     * @title write
      * @param sessionId        A sessionId returned by a call to the icat server.
      * @param investigationIds If present, a comma separated list of investigation id values
      * @param datasetIds       If present, a comma separated list of data set id values or
@@ -688,7 +689,6 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws InternalException
      * @throws NotFoundException
-     * @title write
      * @statuscode 200 To indicate success
      */
     @POST

--- a/src/main/java/org/icatproject/ids/IdsService.java
+++ b/src/main/java/org/icatproject/ids/IdsService.java
@@ -66,7 +66,7 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws InternalException
      * @throws NotFoundException
-     * @summary archive
+     * @title archive
      * @statuscode 200 To indicate success
      */
     @POST
@@ -95,7 +95,7 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InternalException
      * @throws DataNotOnlineException
-     * @summary delete
+     * @title delete
      * @statuscode 200 To indicate success
      */
     @DELETE
@@ -116,7 +116,7 @@ public class IdsService {
      * Return the version of the server
      *
      * @return json string of the form: <samp>{"version":"4.4.0"}</samp>
-     * @summary Version
+     * @title Version
      */
     @GET
     @Path("version")
@@ -158,7 +158,7 @@ public class IdsService {
      * @throws InternalException
      * @throws InsufficientPrivilegesException
      * @throws DataNotOnlineException
-     * @summary getData
+     * @title getData
      * @statuscode 200 To indicate success
      */
     @GET
@@ -208,7 +208,7 @@ public class IdsService {
      * @throws InternalException
      * @throws NotFoundException
      * @throws InsufficientPrivilegesException
-     * @summary getDatafileIds
+     * @title getDatafileIds
      * @statuscode 200 To indicate success
      */
     @GET
@@ -259,7 +259,7 @@ public class IdsService {
      * @throws InternalException
      * @throws NotFoundException
      * @throws DataNotOnlineException
-     * @summary getLink
+     * @title getLink
      * @statuscode 200 To indicate success
      */
     @POST
@@ -283,7 +283,7 @@ public class IdsService {
      * @return a json string.
      * @throws InternalException
      * @throws InsufficientPrivilegesException
-     * @summary getServiceStatus
+     * @title getServiceStatus
      * @statuscode 200 To indicate success
      */
     @GET
@@ -309,7 +309,7 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InsufficientPrivilegesException
      * @throws InternalException
-     * @summary getSize
+     * @title getSize
      * @statuscode 200 To indicate success
      */
     @GET
@@ -347,7 +347,7 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InsufficientPrivilegesException
      * @throws InternalException
-     * @summary getStatus
+     * @title getStatus
      * @statuscode 200 To indicate success
      */
     @GET
@@ -387,7 +387,7 @@ public class IdsService {
      * @throws BadRequestException
      * @throws NotFoundException
      * @throws InternalException
-     * @summary isPrepared
+     * @title isPrepared
      * @statuscode 200 To indicate success
      */
     @GET
@@ -403,7 +403,7 @@ public class IdsService {
      * readOnly status of the server.
      *
      * @return true if readonly, else false
-     * @summary isReadOnly
+     * @title isReadOnly
      * @statuscode 200 To indicate success
      */
     @GET
@@ -418,7 +418,7 @@ public class IdsService {
      * storage. This returns the twoLevel status of the server.
      *
      * @return true if twoLevel, else false
-     * @summary isTwoLevel
+     * @title isTwoLevel
      * @statuscode 200 To indicate success
      */
     @GET
@@ -432,7 +432,7 @@ public class IdsService {
      * Should return "IdsOK"
      *
      * @return "IdsOK"
-     * @summary ping
+     * @title ping
      * @statuscode 200 To indicate success
      */
     @GET
@@ -466,7 +466,7 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws NotFoundException
      * @throws InternalException
-     * @summary prepareData
+     * @title prepareData
      * @statuscode 200 To indicate success
      */
     @POST
@@ -503,7 +503,7 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws NotImplementedException
      * @throws DataNotOnlineException
-     * @summary put
+     * @title put
      * @statuscode 201 When object successfully created
      */
     @PUT
@@ -537,7 +537,7 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws NotImplementedException
      * @throws DataNotOnlineException
-     * @summary putAsPost
+     * @title putAsPost
      * @statuscode 201 When object successfully created
      */
     @POST
@@ -625,7 +625,7 @@ public class IdsService {
      * @throws NotFoundException
      * @throws InternalException
      * @throws InsufficientPrivilegesException
-     * @summary reset
+     * @title reset
      * @statuscode 200 To indicate success
      */
     @POST
@@ -657,7 +657,7 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws InternalException
      * @throws NotFoundException
-     * @summary restore
+     * @title restore
      * @statuscode 200 To indicate success
      */
     @POST
@@ -688,7 +688,7 @@ public class IdsService {
      * @throws InsufficientPrivilegesException
      * @throws InternalException
      * @throws NotFoundException
-     * @summary write
+     * @title write
      * @statuscode 200 To indicate success
      */
     @POST

